tools/git-scm/src/Repository.php

```diff
@@ -0,0 +1,687 @@
+     */
+    public function getTags(): array
+    {
+        $o = $this->run('tag', ['--sort=-creatordate', '--format=%(refname:short) %(objectname)']);
+
+        if (empty($o) || 0 !== $this->exitCode) {
+            return [];
+        }
+
+        if (!isset($this->cache[$i = \md5($o)])) {
+            foreach (\explode("\n", $o) as $line) {
+                if (!empty($line)) {
+                    [$tag, $hash] = \explode(' ', $line, 2);
+                    $this->cache[$i][] = new Tag($this, 'refs/tags/'.$tag, $hash);
+                }
+            }
+        }
+
+        return $this->cache[$i];
+    }
+
+    /**
+     * Returns tag(s) belonging to a revision, if count is 1, the first tag is returned.
+     *
+     * @param string $revision The tag ref name or short name, tag, or a commit hash
+     *
+     * @return null|array<int,Tag>|Tag
+     */
+    public function getTag(string $revision = 'HEAD'): Tag|array|null
+    {
+        $o = $this->run('tag', ['--points-at', $revision, '--format=%(refname:short) %(objectname)']);
+
+        if (empty($o) || 0 !== $this->exitCode) {
+            return null;
+        }
+
+        if (!isset($this->cache[$i = \md5($o)])) {
+            foreach (\explode("\n", $o) as $line) {
+                if (!empty($line)) {
+                    [$tag, $hash] = \explode(' ', $line, 2);
+                    $this->cache[$i][] = new Tag($this, 'refs/tags/'.$tag, $hash);
+                }
+            }
+
+            if (1 <= \count($tags = &$this->cache[$i] ?? [])) {
+                $tags = $tags[0] ?? [];
+            }
+        }
+
+        return $this->cache[$i];
+    }
+
+    /**
+     * Returns a commit object from it given hash.
+     */
+    public function getCommit(string $commitHash): Commit
+    {
+        return $this->cache[$commitHash] ??= new Commit($this, $commitHash);
+    }
+
+    /**
+     * Returns the current commit.
+     */
+    public function getLastCommit(): ?Commit
+    {
+        $o = $this->run('log', ['--pretty=format:%H', '-n', 1]);
+
+        if (empty($o) || 0 !== $this->exitCode) {
+            return null;
+        }
+
+        return $this->getCommit(\trim($o));
+    }
+
+    /**
+     * @param null|array<int,string>|string $revisions a list of revisions or null if you want all history
+     * @param null|array<int,string>|string $paths     paths to filter on
+     * @param null|int                      $offset    start list from a given position
+     * @param null|int                      $limit     limit number of fetched elements
+     */
+    public function getLog(array|string $revisions = null, array|string $paths = null, int $offset = null, int $limit = null): Log
+    {
+        return new Log($this, $revisions, $paths, $offset, $limit);
+    }
+
+    /**
+     * Returns the global default git user.
+     */
+    public function getAuthor(): ?Commit\Identity
+    {
+        if (!isset($this->cache['author'])) {
+            $data = $this->runConcurrent([
```

Comment:
I think the getAuthor method should not thrown an exception but return null instead if execution fails